Disable RPC profiling for kineto profilers

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #76234

RPC profiling is only enabled when the profiler is of legacy type.

Differential Revision: [D35484579](https://our.internmc.facebook.com/intern/diff/D35484579/)

rohan-varma committed:
Disable RPC profiling for kineto profilers

RPC profiling is only enabled when the profiler is of legacy type.

Differential Revision: [D35484579](https://our.internmc.facebook.com/intern/diff/D35484579/)

[ghstack-poisoned]

H-Huang commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
```diff
@@ -2186,6 +2206,11 @@ def _run_test_profiler_with_async_rpc_udf(self):
+        )
 
     @dist_init
     def test_profiler_with_async_rpc_udf(self):
```
Unrelated, curious about this test. Why does this and the test below it call the same func?

rohan-varma replied:
Hmm the single threaded one was supposed to test RPC with num_worker_threads=1. Perhaps that went away in the tensorpipe refactor or somehow?

H-Huang reviewed (approved):
LGTM, stamp to unblock

rohan-varma committed:
Update on "Disable RPC profiling for kineto profilers"


RPC profiling is only enabled when the profiler is of legacy type.

Differential Revision: [D35484579](https://our.internmc.facebook.com/intern/diff/D35484579/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Disable RPC profiling for kineto profilers"


RPC profiling is only enabled when the profiler is of legacy type.

Differential Revision: [D35484579](https://our.internmc.facebook.com/intern/diff/D35484579/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Disable RPC profiling for kineto profilers"


RPC profiling is only enabled when the profiler is of legacy type.

Differential Revision: [D35484579](https://our.internmc.facebook.com/intern/diff/D35484579/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Disable RPC profiling for kineto profilers"


RPC profiling is only enabled when the profiler is of legacy type.

Differential Revision: [D35484579](https://our.internmc.facebook.com/intern/diff/D35484579/)

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge this please
